pyccl/halos/halo_model.py

```diff
@@ -30,46 +31,47 @@ class HMCalculator(object):
+        mass_function (:class:`~pyccl.halos.hmfunc.MassFunc`): a mass
             function object.
-        hbias (:class:`~pyccl.halos.hbias.HaloBias`): a halo bias
+        halo_bias (:class:`~pyccl.halos.hbias.HaloBias`): a halo bias
             object.
         mass_def (:class:`~pyccl.halos.massdef.MassDef`): a mass
             definition object.
-        log10M_min (float): logarithmic mass (in units of solar mass)
-            corresponding to the lower bound of the integrals in
-            mass. Default: 8.
-        log10M_max (float): logarithmic mass (in units of solar mass)
-            corresponding to the upper bound of the integrals in
-            mass. Default: 16.
-        nlog10M (int): number of samples in log(Mass) to be used in
-            the mass integrals. Default: 128.
+        lM_min, lM_max (float): lower and upper integration bounds
+            of logarithmic (base-10) mass (in units of solar mass).
+            Default range: 8, 16.
+        nlM (int): number of uniformly-spaced samples in log(Mass)
+            to be used in the mass integrals. Default: 128.
         integration_method_M (string): integration method to use
             in the mass integrals. Options: "simpson" and "spline".
             Default: "simpson".
-        k_min (float): some of the integrals solved by this class
+        k_norm (float): some of the integrals solved by this class
             will often be normalized by their value on very large
             scales. This parameter (in units of inverse Mpc)
             determines what is considered a "very large" scale.
             Default: 1E-5.
     """
-    def __init__(self, cosmo, massfunc, hbias, mass_def,
-                 log10M_min=8., log10M_max=16.,
-                 nlog10M=128, integration_method_M='simpson',
-                 k_min=1E-5):
+    @warn_api(pairs=[("mass_function", "massfunc"), ("halo_bias", "hbias"),
+                     ("lM_min", "log10M_min"), ("lM_max", "log10M_max"),
+                     ("nlM", "nlog10M"), ("k_norm", "k_min")])
+    def __init__(self, cosmo, *, mass_function, halo_bias, mass_def,
+                 lM_min=8., lM_max=16., nlM=128,
+                 integration_method_M='simpson', k_norm=1E-5):
         self._rho0 = rho_x(cosmo, 1., 'matter', is_comoving=True)
-        if not isinstance(massfunc, MassFunc):
-            raise TypeError("massfunc must be of type `MassFunc`")
-        self._massfunc = massfunc
-        if not isinstance(hbias, HaloBias):
-            raise TypeError("hbias must be of type `HaloBias`")
-        self._hbias = hbias
-        self._mdef = mass_def
-        self._prec = {'log10M_min': log10M_min,
-                      'log10M_max': log10M_max,
-                      'nlog10M': nlog10M,
+        if not isinstance(mass_function, MassFunc):
+            raise TypeError("mass_function must be of type `MassFunc`")
+        self.mass_function = mass_function
```

Comment:
we had intended for these attributes to be private (i.e. not meant to be accessed by the user), so I'd keep the `_`

Reply:
OK. Why though? Someone might want to do something like `isinstance(hmc.mass_function, MassFuncTinker08)`.